src/ResourceManagement/ContainerRegistry/RegistriesImpl.cs

```diff
@@ -5,117 +5,168 @@ namespace Microsoft.Azure.Management.ContainerRegistry.Fluent
-                var containerService = await Task.WhenAll(resourceGroups.Select(async (rg) => await ListInnerByGroupAsync(rg.Name, cancellation)));
-                return containerService.SelectMany(x => x);
-            }, WrapModel, cancellationToken);
+                return new ReadOnlyCollection<Models.RegistryUsage>(registryUsageListResultInner.Value);
+            }
+            else
+            {
+                return new List<RegistryUsage>().AsReadOnly();
+            }
         }
 
+        ///GENMHASH:601511CD9B176CC455F825E8F050ED1E:B8E6D924E302EBC698D644CC87321B9F
+        public IRegistryCredentials GetCredentials(string resourceGroupName, string registryName)
+        {
+            return new RegistryCredentialsImpl(Extensions.Synchronize(() => this.Inner.ListCredentialsAsync(resourceGroupName, registryName)));
+        }
 
-        protected async override Task<IPage<Models.RegistryInner>> ListInnerAsync(CancellationToken cancellationToken)
+        ///GENMHASH:2B15DD784A432B193101991B2A395D8E:F1E4CE445A91F87D1EAE7E4DC4FE5EF5
+        public async Task<Microsoft.Azure.Management.ContainerRegistry.Fluent.IRegistryCredentials> GetCredentialsAsync(string resourceGroupName, string registryName, CancellationToken cancellationToken = default(CancellationToken))
         {
-            return await Inner.ListAsync(cancellationToken);
+            return new RegistryCredentialsImpl(await this.Inner.ListCredentialsAsync(resourceGroupName, registryName, cancellationToken));
         }
 
-        protected async override Task<IPage<Models.RegistryInner>> ListInnerNextAsync(string nextLink, CancellationToken cancellationToken)
+        ///GENMHASH:9901FC42F194D4C02E5A4EF45224492B:B1F8C403ADAD3A5658412B1A5E6AFFD8
+        public IRegistryCredentials RegenerateCredential(string resourceGroupName, string registryName, AccessKeyType accessKeyType)
         {
-            return await Inner.ListNextAsync(nextLink, cancellationToken);
+            return new RegistryCredentialsImpl(Extensions.Synchronize(() => this.Inner.RegenerateCredentialAsync(resourceGroupName, registryName, accessKeyType.ToPasswordName())));
         }
 
-        protected async override Task<IPage<Models.RegistryInner>> ListInnerByGroupAsync(string groupName, CancellationToken cancellationToken)
+        ///GENMHASH:5B0E9060E6A010D9AEC0DEE8C817853F:81A8B8E3661A768A3015CC178EFC3516
+        public async Task<Microsoft.Azure.Management.ContainerRegistry.Fluent.IRegistryCredentials> RegenerateCredentialAsync(string resourceGroupName, string registryName, AccessKeyType accessKeyType, CancellationToken cancellationToken = default(CancellationToken))
         {
-            return await this.Inner.ListByResourceGroupAsync(groupName, cancellationToken);
+            return new RegistryCredentialsImpl(await this.Inner.RegenerateCredentialAsync(resourceGroupName, registryName, accessKeyType.ToPasswordName()));
         }
```

Comment:
please pass cancellation token to async call

Reply:
done